index.js

```diff
@@ -66,152 +62,109 @@ function releaseConnectionToPool(release) {
 
 function decorateWithQueryRowsAsync(env, client) {
   return Object.assign(client, {
-    queryRowsAsync: (query, args) => queryWithCtxAsync(env, client, query, args).then(getRows)
+    queryRowsAsync: (query, args) => client.queryAsync(query, args).then(res => res.rows)
   })
 }
 
-function executeQueryRowsAsync(env, connector, query, args) {
-  return using(getConnection(env, connector), function (connection) {
-    return connection.queryRowsAsync(query, args)
-  })
+function executeQueryRows(env, connector, query, args) {
+  return using(getConnection(env, connector), connection =>
+    connection.queryRowsAsync(query, args)
+  )
 }
 
-function getRows(res) {
-  return res.rows
-}
-
-function queryWithCtxAsync(env, client, query, args) {
-  if (_.isObject(query) && query[env.queryValuesKey] && Array.isArray(args) && args.length > 0) {
-    throw new Error('Both query.values and args were passed to query. Please use only one of them.')
-  }
-
-  return client.queryAsync(query[env.queryTextKey] || query, query[env.queryValuesKey] || args)
+function executeQuery(env, connector, query, args) {
+  return using(getConnection(env, connector), connection =>
+    connection.queryAsync(query, args)
+  )
 }
 
 function constructLockingBeginStatement(involvedTables) {
-  var lockSql = 'LOCK TABLE {table} IN SHARE ROW EXCLUSIVE MODE'
-  var statements = involvedTables.map(function(table) {
-    return stringTemplate(lockSql, { table: table })
-  })
-  statements.unshift('BEGIN')
-  return statements.join(';')
+  const statements = involvedTables.map(table =>
+    `LOCK TABLE ${table} IN SHARE ROW EXCLUSIVE MODE`
+  )
+  return ['BEGIN'].concat(statements).join(';')
 }
 
-function createUpsertCTE(table, idField, args) {
-  var insert = args.insert
-  var update = args.update
+function createMultipleInsertCTE(insert) {
```

Comment:
I think these utilities should be removed and the release should be a major version upgrade.

Reply:
Bumped version. Will keep the createMultipleInsertCTE util for the moment as it could still be used for some specific cases, but will consider removing it or entirely refactoring it in the future.